snapcraft/_options.py

```diff
@@ -124,7 +124,68 @@ def _get_platform_architecture():
     return architecture
 
 
-class ProjectOptions:
+class ProjectInfo:
+    """Information gained from the snap's snapcraft.yaml file."""
+
+    def __init__(self, *,
+                 name: str, version: str,
+                 summary: str, description: str, confinement: str) -> None:
+        self.__name = name
+        self.__version = version
+        self.__summary = summary
+        self.__description = description
+        self.__confinement = confinement
+
+    @property
+    def name(self):
+        """Returns the name of the snap.
+        """
+        return self.__name
```

Comment:
I think these should be actual functions. Otherwise they look like attributes until someone tries to set them. Functions provide a more obvious and easy-to-use API.

Reply:
How is this different to how we use properties elsewhere?